packages/genesys-spark-components/src/components/beta/gux-side-panel/gux-modal-side-panel.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Listen,
+  Method,
+  Prop,
+  Watch
+} from '@stencil/core';
+import { GuxSidePanelSize } from './gux-side-panel.types';
+import { randomHTMLId } from '@utils/dom/random-html-id';
+import { hasSlot } from '@utils/dom/has-slot';
+import { trackComponent } from '@utils/tracking/usage';
+
+@Component({
+  tag: 'gux-modal-side-panel-beta',
+  styleUrl: './gux-modal-side-panel.scss',
+  shadow: true
+})
+export class GuxModalSidePanel {
+  @Element()
+  private root: HTMLElement;
+
+  @Prop({ mutable: true })
+  open: boolean = false;
+
+  @Prop()
+  size: GuxSidePanelSize = 'medium';
+
+  private dialogElement: HTMLDialogElement;
+  @Watch('open')
+  syncOpenState() {
+    if (this.open) {
+      this.dialogElement.showModal();
+    } else {
+      this.dialogElement.close();
+      this.modalSidePanelDismiss.emit();
+    }
+  }
+
+  @Event()
+  modalSidePanelDismiss: EventEmitter<void>;
```

Comment:
Optional but I think `guxdismiss` might be better as it would be consistent with other components in the repo.